dolphin/tests/unit/api/v1/test_alert.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from unittest import mock
+
+from oslo_utils import importutils
+
+from dolphin import exception
+from dolphin.tests.unit.api import fakes
+
+
+class AlertControllerTestCase(unittest.TestCase):
+    ALERT_CONTROLLER_CLASS = 'dolphin.api.v1.alert.AlertController'
+
+    @mock.patch('dolphin.alert_manager.rpcapi.AlertAPI', mock.Mock())
+    def _get_alert_controller(self):
+        alert_controller_class = importutils.import_class(
+            self.ALERT_CONTROLLER_CLASS)
+        alert_controller = alert_controller_class()
+        return alert_controller
+
+    @mock.patch('dolphin.db.storage_get', mock.Mock())
+    @mock.patch('dolphin.db.alert_source_update')
+    @mock.patch('dolphin.db.alert_source_get')
+    def test_put_v3_config_create_success(self, mock_alert_source_get,
+                                          mock_alert_source_update):
+        req = fakes.HTTPRequest.blank('/storages/fake_id/alert-source')
+        fake_storage_id = 'abcd-1234-5678'
+        expected_alert_source = {'storage_id': 'abcd-1234-5678',
+                                 'host': '127.0.0.1',
+                                 'community_string': None,
+                                 'version': 'snmpv3',
+                                 'engine_id': '800000d30300000e112245',
+                                 'security_level': None,
+                                 'username': 'test1',
+                                 'auth_protocol': 'md5',
+                                 'privacy_protocol': 'des',
+                                 "created_at": None,
```

Comment:
IMO, created_at in output_alert_source is not None, so will line 61 success?

Reply:
Updated